src/utils.rs

```diff
@@ -0,0 +1,54 @@
+#[cfg(test)]
+use core::fmt::{self, Write};
+
+/// Sink that will return an error when bytes are written to it that do not match `expected`.
+#[cfg(test)]
+#[allow(dead_code)]
+pub(crate) struct WriteCompare<'a> {
+    expected: &'a str,
+    remainder: &'a str,
+}
+
+#[cfg(test)]
+impl<'a> WriteCompare<'a> {
+    pub(crate) fn new(expected: &'a str) -> Self {
+        Self { expected, remainder: expected }
+    }
+}
+
+#[cfg(test)]
+impl<'a> Write for WriteCompare<'a> {
+    fn write_str(&mut self, s: &str) -> fmt::Result {
+        if let Some(remainder) = self.remainder.strip_prefix(s) {
+            self.remainder = remainder;
+            Ok(())
+        } else {
+            #[cfg(feature = "std")]
+            eprintln!(
+                "formatting difference: `(left == right)`\n  left: `\"{}{}(...)\"`\n right: `\"{}\"`",
```

Comment:
I suggest using `{:?}` here so formatting characters like newline and tab are escaped. You can remove the explicit `"` in the string.

Reply:
I didn't know that! 👍
The `"`'s are to imitate the format of the `assert_eq!` macro.